sdk/ml/azure-ai-ml/azure/ai/ml/entities/_endpoint/batch_endpoint.py

```diff
@@ -54,7 +54,7 @@ def __init__(
         auth_mode: str = AAD_TOKEN_YAML,
         description: str = None,
         location: str = None,
-        defaults: Dict[str, str] = None,
+        defaults: Dict[str, str] = {},
```

Comment:
Default values for mutable types should always be `None`, as their value is instantiated only once.

Reply:
I'm not fully understanding the problem tho, our public doc will look at our type hints, here is `Dict[str,str]`, which is what we expect the caller to pass. 

Is the problem that, user can not pass `defaults` and also not `default_deployment` which makes `self.defaults == None`? Then we can break the logic s.t. `self.defaults` is always at least `{}`.

Reply:
@qusayalsaifi Still looking for a change here. We cannot make this change. 
Here's a simple example:
```
class Foo:
	def __init__(self, defaults = {}):
		self.defaults = defaults

a = Foo()
print(a.defaults) # prints {}
a.defaults['key1'] = 'value1'
print(a.defaults) # prints {'key1': 'value1'}

b = Foo()
print(b.defaults) # prints {'key1': 'value1'}
```

If you want `self.defaults` to be always instantiated, you must do it in the ctor itself and not as a default value.

Reply:
@qusayalsaifi : Could you share more about this issue? Do you have the user code so we can repro?

I am hesitant to change the default since it could be a breaking change, and also @Man-MSFT has a point.

But I would like to understand better the root cause to find the right solution.